### PR TITLE
Generate API types using code generation

### LINKED ISSUE #311: Use code generation to create API types
## Summary
Similar to [go-algorand-sdk](https://github.com/algorand/go-algorand-sdk) and [java-algorand-sdk](https://github.com/algorand/java-algorand-sdk), this SDK can use code generation to create request and response types for the algod and indexer APIs.

## Scope
Create and run code generation templates to generate TypeScript files for algod and indexer responses and requests.

## Urgency
Required to finish the TypeScript migration.

### PR DESCRIPTION
**Pull request by jdtzmn**
This PR includes a code template for API types and generated response types for the algod and indexer APIs.

Closes #311.

### CODE REVIEW

File: src/types/blockHeader.d.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+/**
```
Comment (jasonpaulos) [inline src/types/blockHeader.d.ts:1]: I think this file ending should be `.ts`.

File: src/client/v2/algod/models/types.ts [hunk 2]
```diff
@@ -0,0 +1,1390 @@
...
+    protocolVersion,
+    round,
+    sources,
+    txns,
+  }: {
+    accounts: Account[];
+    apps: Application[];
+    latestTimestamp: number;
+    protocolVersion: string;
+    round: bigint;
```
Comment (jasonpaulos) [inline src/client/v2/algod/models/types.ts:672]: I wonder if we should make the `uint64` and `integer` API types map to `number | bigint`? They should both get encoded the same way by the SDK, so I think this change would make the types more flexible to "input" types like this that users create.

**Review by jasonpaulos — COMMENTED** [review]
Mostly looks good!

I noticed that the new generated types aren't used yet, but I think we can make a separate issue for that.

File: src/client/v2/algod/models/types.ts (hunk 2 above)
Comment (jdtzmn) [reply src/client/v2/algod/models/types.ts:672]: Yes, I think that makes sense.

**Review by jasonpaulos — APPROVED** [review]

**Pull request merged**